### PR TITLE
Add concepts page to docs

### LINKED ISSUE #2433: Add Concepts page to Seldon Core docs
A page to provide information on concepts used in Seldon Core and definitions for the most commonly used technical terms.

### PR DESCRIPTION
**Pull request by glindsell**
Signed-off-by: glindsell <gl@seldon.io>

**Which issue(s) this PR fixes**:
Fixes #2433

```release-note
Add Concepts page to Seldon Core docs
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **glindsell**
You can assign the PR to them by writing `/assign @glindsell` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Sep 14 14:24:08 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2434/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2434 --build=2

**Comment by seldondev** [comment]
Mon Sep 14 14:24:08 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2434/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2434 --build=1

File: doc/source/reference/concepts.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Seldon Core Concepts
+
+This page is a work in progress to provide concepts related to Seldon Core.
+
+This work is ongoing and we welcome feedback
+
+## Workflow
+#### Conceptual overview of workflows
```
Comment (adriangonz) [inline doc/source/reference/concepts.md:8]: Would the `####` header be a new subsection or are you using it as a subtitle? It's a bit annoying that markdown doesn't have the concept of subtitles. Maybe using `>` could help?

File: doc/source/reference/concepts.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+A workflow groups the components of a machine learning system into a logical pipeline. Sets of related components are grouped into graphs. The workflow contains the configuration of the components and the definitions of the inputs and outputs of the system, and of each component.
```
Comment (adriangonz) [inline doc/source/reference/concepts.md:9]: Would a workflow also encompass the training side or just inference? If it's just inference, we could also add that we also refer to this one as an "inference graph" or "ml deployment".

File: doc/source/reference/concepts.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    
+-   Transformer
+    
+-   Output Transformer
+
+A program that performs one step in the workflow.
+
+## Machine Learning Deployment
+#### Conceptual overview of machine learning deployments
+A machine learning deployment refers to a predictive model in the Seldon ecosystem of a type associated with Seldon (Seldon Deployments) or Kubeflow (Inference Services).
```
Comment (adriangonz) [inline doc/source/reference/concepts.md:28]: An ML deployment could also encompass an inference graph (i.e. not just a predictive model). Seldon Core only focuses on Seldon Deployments, so maybe we could remove the last bit?

File: doc/source/reference/concepts.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+## Model
+#### Conceptual overview of models
+A component within a machine learning deployment which holds the representation of learned information from the training data.
+
+## Language Wrapper
+#### Conceptual overview of language wrappers
+Enables cross language and/or runtime interoperability of a model with a particular programming language.
```
Comment (adriangonz) [inline doc/source/reference/concepts.md:36]: I would also add here that a language wrapper is an inference server (i.e. it responds to requests, it communicates with a service orchestrator, etc.)

File: doc/source/reference/concepts.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+## Pre-packaged Inference Server
+#### Conceptual overview of pre-packaged inference servers
+A pre-packaged inference server is one of:
```
Comment (adriangonz) [inline doc/source/reference/concepts.md:40]: We can also clarify that pre-packaged inference servers come built into Seldon Core, to allow users to go easily from artifact (i.e. serialised model) to ML deployment regardless of toolkit. 

We can also link to this docs page for users looking for instructions on how to create their own "pre-packaged" inference servers: 

- https://docs.seldon.io/projects/seldon-core/en/latest/servers/overview.html
- https://docs.seldon.io/projects/seldon-core/en/latest/servers/custom.html

File: doc/source/reference/concepts.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    
+Servers which can be used to deploy a trained model.
+
+## Graph
+#### Conceptual overview of graphs
+A graph represents machine learning components as nodes with edges representing the inputs and outputs of an operation being passed from one component to the next.
+
+## Request
+#### Conceptual overview of requests
+A request represents a single call to a model for a prediction. The request will be a payload which holds prediction data (often in the form of an array) passed over a particular protocol.
```
Comment (adriangonz) [inline doc/source/reference/concepts.md:58]: We can clarify here that the request is expected to follow a particular format.

**Comment by seldondev** [comment]
Thu Sep 17 10:38:51 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2434/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2434 --build=3

**Comment by seldondev** [comment]
Thu Sep 17 10:39:01 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2434/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2434 --build=4

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[adriangonz](https://github.com/SeldonIO/seldon-core/pull/2434#issuecomment-694153503 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-694153503/SeldonIO/seldon-core/blob/master/OWNERS)~~ [adriangonz]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**